### PR TITLE
Prolink RPC Link Format Implementation

### PR DESCRIPTION
**Pull request by spencerstock**
## What changed? Why?

Implements the Compressed RPC Link Format (Prolink) specification for encoding wallet RPC requests into compact, URL-safe payloads suitable for QR codes, NFC tags, and deep links.

Matches the spec defined here: https://github.com/ethereum/ERCs/pull/1261

### Core Implementation
- Protocol Buffers v3 serialization for binary message encoding
- Optional Brotli compression layer (quality 5, window 22 bits)
- Base64url encoding without padding for URL safety
- Three standardized shortcuts for optimized encoding

### Shortcuts
**Shortcut 0 (Generic JSON-RPC)**: Universal fallback for any RPC method
**Shortcut 1 (wallet_sendCalls)**: Optimized for EIP-5792 with specialized encodings for:
- ERC20 transfers (60-70% size reduction)
- Native transfers
- Generic call batches

**Shortcut 2 (wallet_sign)**: Optimized for EIP-7871 with specialized encodings for:
- SpendPermission signatures
- ReceiveWithAuthorization signatures
- Generic EIP-712 typed data

### Features
- Canonical encodings for addresses (20 bytes, lowercase normalized)
- Minimal big-endian encoding for amounts (no leading zeros)
- Capabilities extension point for metadata (ERC-8026 dataCallback support)
- Protocol versioning and backward compatibility
- Error handling with validation and rejection of malformed data

## How was this tested?

Test coverage for:
- End-to-end encoding/decoding roundtrips for all shortcuts
- Base64url encoding/decoding with edge cases (padding, invalid characters)
- Amount encoding (zero values, minimal encoding, leading zero rejection)
- Address encoding (length validation, normalization)
- ERC20 transfer detection (selector, data length, value validation)
- Native transfer detection (empty data, non-zero value)
- Brotli compression (Node.js and browser environments)
- Protocol versioning and error cases
- Capabilities encoding/decoding (JSON values as UTF-8 bytes)

## How can reviewers manually test these changes?

1. Navigate to the prolink playground page at `/prolink-playground`
2. Test ERC20 transfer encoding:
   - Enter USDC contract, recipient, and amount
   - Click "Generate Prolink" 
   - Click "Decode" to verify roundtrip accuracy
3. Test native transfer encoding:
   - Switch to native transfer tab
   - Enter recipient and ETH amount
   - Verify encoding/decoding roundtrip
4. Test wallet_sign encoding:
   - Switch to SpendPermission tab
   - Fill in permission parameters
   - Verify signature request encoding
5. Compare payload sizes: check that compressed payloads are 60-80% smaller than raw JSON

## Demo/screenshots

https://github.com/user-attachments/assets/c4cdb825-5ecc-445f-becc-116e3289305c

### CODE REVIEW

File: packages/account-sdk/src/index.ts [hunk 1]
```diff
@@ -39,3 +39,8 @@ export type {
   SubscriptionStatus,
   SubscriptionStatusOptions,
 } from './interface/payment/index.js';
+
+export { decodeProlink, encodeProlink } from './interface/public-utilities/prolink/index.js';
+export type { ProlinkDecoded, ProlinkRequest } from './interface/public-utilities/prolink/index.js';
+
+export { showProlinkDialog } from './ui/ProlinkDialog/index.js';
```
[the remark this answers is not in the document]
Comment (spencerstock) [reply packages/account-sdk/src/index.ts]: removed showProlinkDialog

**Comment by cb-jake** [comment]
Whats the prompt to store this in the account sdk? This feels like a separate package that account sdk could likely consume?

File: packages/account-sdk/src/interface/public-utilities/prolink/utils/encoding.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+/**
+ * Convert hex string to bytes
+ * @param hex - Hex string (with or without 0x prefix)
+ * @returns Bytes
+ */
+export function hexToBytes(hex: string): Uint8Array {
+  const normalized = hex.toLowerCase().replace(/^0x/, '');
+  const bytes = new Uint8Array(normalized.length / 2);
```
Comment (jxom) [inline packages/account-sdk/src/interface/public-utilities/prolink/utils/encoding.ts]: If hex is `"0x123"`, then `normalized.length = 3`, and `bytes = new Uint8Array(1.5)` becomes `new Uint8Array(1)`, so we will lose data here.

I would recommend using `Bytes`, `Base64`, etc from [Ox](https://oxlib.sh) for all these utilities fwiw since they are battle-tested and more performant.

Comment (spencerstock) [reply packages/account-sdk/src/interface/public-utilities/prolink/utils/encoding.ts]: You're right, odd length hex is broken here. Adding test cases for coverage then applying a fix.

File: examples/testapp/next.config.mjs [hunk 3]
```diff
@@ -7,4 +9,21 @@ export default {
...
+
+      // Ignore node:zlib imports in client-side code
+      config.plugins.push(
+        new webpack.IgnorePlugin({
+          resourceRegExp: /^node:zlib$/,
+        })
+      );
+    }
+    return config;
+  },
```
Comment (stephancill) [inline examples/testapp/next.config.mjs]: assuming consumers of our SDK don't have to do this?

File: packages/account-sdk/src/ui/ProlinkDialog/ProlinkDialog.tsx
Comment (stephancill) [inline packages/account-sdk/src/ui/ProlinkDialog/ProlinkDialog.tsx]: is this used in the playground anywhere to see what it looks like?

File: examples/testapp/next.config.mjs (hunk 3 above)
Comment (spencerstock) [reply examples/testapp/next.config.mjs]: Updated to Implement environment specific exports (and removed this)

File: packages/account-sdk/src/interface/public-utilities/prolink/shortcuts/sign.ts [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  if (type === '0x1') return '0x01';
+  return type;
+}
+
+/**
+ * Detect if typed data is a SpendPermission
+ */
+function detectSpendPermission(typedData: TypedData): boolean {
+  return (
+    typedData.primaryType === 'SpendPermission' && typedData.domain.verifyingContract !== undefined
```
[the remark this answers is not in the document]
Comment (spencerstock) [reply packages/account-sdk/src/interface/public-utilities/prolink/shortcuts/sign.ts]: I don't think so. Another implementation of a permission manager contract could still be valid in this context

File: packages/account-sdk/src/ui/ProlinkDialog/ProlinkDialog.tsx
Comment (spencerstock) [reply packages/account-sdk/src/ui/ProlinkDialog/ProlinkDialog.tsx]: I removed this for now - we may add it later if requested by devs

**Comment by cb-jake** [comment]
@spencerstock, I'm not convinced this belongs in the SDK. I could see this being a separate package that the core SDK can consume. Would be curious if any others feel the same or differently

**Review by fan-zhang-sv — DISMISSED** [review]
agree looking from this PR this doesn't seem to be the best place for prolink

but from the sync w builder team this morning, sounds like
* this is not final, and is for unblocking devs in this interim period before Bruno's standalone lib is live
* we may migrate to import from standalone lib and convert this into a branded flow (?)

stamping to unblock, let's do quick iteration on our offering

**Review by fan-zhang-sv — APPROVED** [review]

**Review by stephancill — APPROVED** [review]
